src/FsAutoComplete.Core/AdaptiveExtensions.fs

```diff
@@ -600,17 +664,27 @@ module AsyncAVal =
   /// </summary>
   let map (mapping: 'a -> CancellationToken -> Task<'b>) (input: asyncaval<'a>) =
     let mutable cache: option<RefCountingTaskCreator<'b>> = None
+    let mutable dataCache = ValueNone
```

Comment:
Add caching support to map/bind. If we get the same value from `input`, we don't need to recall the mapping function.